pkg/client/client_test.go

```diff
@@ -1430,6 +1430,130 @@ var _ = Describe("Client", func() {
+				Expect(err).NotTo(HaveOccurred())
+				dep3 := &appsv1.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "deployment-3",
+					},
+					Spec: appsv1.DeploymentSpec{
+						Selector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{"foo": "bar"},
+						},
+						Template: corev1.PodTemplateSpec{
+							ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"foo": "bar"}},
+							Spec:       corev1.PodSpec{Containers: []corev1.Container{{Name: "nginx", Image: "nginx"}}},
+						},
+					},
+				}
+				dep3, err = clientset.AppsV1().Deployments(ns).Create(dep3)
+				Expect(err).NotTo(HaveOccurred())
+				dep4 := &appsv1.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "deployment-4",
+					},
+					Spec: appsv1.DeploymentSpec{
+						Selector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{"foo": "bar"},
+						},
+						Template: corev1.PodTemplateSpec{
+							ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"foo": "bar"}},
+							Spec:       corev1.PodSpec{Containers: []corev1.Container{{Name: "nginx", Image: "nginx"}}},
+						},
+					},
+				}
+				dep4, err = clientset.AppsV1().Deployments(ns).Create(dep4)
+				Expect(err).NotTo(HaveOccurred())
+
+				cl, err := client.New(cfg, client.Options{})
+				Expect(err).NotTo(HaveOccurred())
+
+				By("listing 1 deployment when limit=1 is used")
+				deps := &appsv1.DeploymentList{}
+				err = cl.List(context.Background(), deps,
+					client.Limit(1),
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(deps.Items).NotTo(BeEmpty())
+				Expect(1).To(Equal(len(deps.Items)))
+				Expect(deps.Continue).NotTo(BeEmpty())
+				actual := deps.Items[0]
+				Expect(actual.Name).To(Equal(dep1.Name))
+
+				continueToken := deps.Continue
+
+				By("listing the next deployment when previous continuation token is used and limit=1")
+				deps = &appsv1.DeploymentList{}
+				err = cl.List(context.Background(), deps,
+					client.Limit(1),
+					client.Continue(continueToken),
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(deps.Items).NotTo(BeEmpty())
+				Expect(1).To(Equal(len(deps.Items)))
+				Expect(deps.Continue).NotTo(BeEmpty())
+				actual = deps.Items[0]
+				Expect(actual.Name).To(Equal(dep2.Name))
+
+				continueToken = deps.Continue
+
+				By("listing the 2 remaining deployments when previous continuation token is used without a limit")
+				deps = &appsv1.DeploymentList{}
+				err = cl.List(context.Background(), deps,
+					client.Continue(continueToken),
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(deps.Items).NotTo(BeEmpty())
+				Expect(2).To(Equal(len(deps.Items)))
+				Expect(deps.Continue).To(BeEmpty())
+				first := deps.Items[0]
+				Expect(first.Name).To(Equal(dep3.Name))
+				second := deps.Items[1]
+				Expect(second.Name).To(Equal(dep4.Name))
+
+				deleteDeployment(dep1, ns)
+				deleteDeployment(dep2, ns)
+				deleteDeployment(dep3, ns)
+				deleteDeployment(dep4, ns)
+
+				close(done)
```

Comment:
no need for a done channel unless you're doing async tests (which this isn't)

Reply:
Same as above. Most tests do this.